content/news/2022-11-05-bevy-0.9/index.md

```diff
@@ -0,0 +1,1088 @@
+
+```rust
+// Old
+fn some_system(world: &mut World) {
+  let mut state: SystemState<(Res<Time>, Query<&mut Transform>)> =
+      SystemState::new(&mut world);
+  let (time, mut transforms) = state.get_mut(world);
+}
+
+// New
+fn some_system(world: &mut World, state: &mut SystemState<(Res<Time>, Query<&mut Transform>)>) {
+  let (time, mut transforms) = state.get_mut(world);
+}
+```
+
+[`QueryState`] can also be used to cache individual queries:
+
+```rust
+// Old
+fn some_system(world: &mut World) {
+  let mut transforms = world.query::<&Transform>();
+  for transform in transforms.iter(world) {
+  }
+}
+
+// New
+fn some_system(world: &mut World, transforms: &mut QueryState<&Transform>) {
+  for transform in transforms.iter(world) {
+  }
+}
+```
+
+[`Local`] can be used to store local data inside of the exclusive system:
+
+```rust
+// Old
+#[derive(Resource)]
+struct Counter(usize);
+fn some_system(world: &mut World) {
+  let mut counter = world.resource_mut::<Counter>();
+  counter.0 += 1;
+}
+
+// New
+fn some_system(world: &mut World, mut counter: Local<usize>) {
+  *counter += 1;
+}
+```
+
+[`World`]: https://docs.rs/bevy/0.9.0/bevy/ecs/world/struct.World.html
+[`QueryState`]: https://docs.rs/bevy/0.9.0/bevy/ecs/query/struct.QueryState.html
+[`SystemState`]: https://docs.rs/bevy/0.9.0/bevy/ecs/system/struct.SystemState.html
+[`Local`]: https://docs.rs/bevy/0.9.0/bevy/ecs/system/struct.Local.html
+
+## Bevy ECS Now Uses GATS!
+
+<div class="release-feature-authors">authors: @BoxyUwU</div>
+
+Rust 1.65.0 [stabilized GATs](https://blog.rust-lang.org/2022/11/03/Rust-1.65.0.html) (Generic Associated Types), which enabled us to significantly simplify Bevy ECS Query internals.
+
+For awhile now, Bevy ECS has been hacking around the lack of GATs with a complicated nest of traits (`WorldQuery`, `WorldQueryGats` (a "lack of real GATs hack" trait), and `Fetch`).
+
+In **Bevy 0.9**, we now have a single [`WorldQuery`] trait! This makes Bevy ECS much easier to maintain, extend, debug, document, and understand.
+
+[`WorldQuery`]: https://docs.rs/bevy/0.9.0/bevy/ecs/query/trait.WorldQuery.html
+
+## Derive Resource
+
+<div class="release-feature-authors">authors: @devil-ira</div>
+
+The [`Resource`] trait is now no longer automatically implemented for all types. It must be derived:
+
+```rust
+#[derive(Resource)]
+struct Counter(usize);
+```
+
+This change was made on the tail of [making the same decision](/news/bevy-0-6/#the-new-component-trait-and-derive-component) for [`Component`] types. In short:
+
+1. Auto-implementing [`Resource`] for every type made it very easy to accidentally insert the "wrong" value, such as inserting the constructor function pointer instead of the value itself:
+
+    ```rust
+    struct Counter(usize);
+    // This inserts the constructor function pointer as a resource!
+    // Weird and confusing!
+    app.insert_resource(Counter);
```

Comment:
For people new to bevy its not clear what is the right way to do this. Maybe include the right way to use `insert_resource` in the example of the new `#[derive(Resource)]`.